### PR TITLE
Add tardies dashboard

### PR DESCRIPTION
**Pull request by edavidsonsawyer**
# Who is this PR for?
School Principals and Vice Principals who want to track tardies schoolwide and by homeroom

# What problem does this PR fix?
Lack of visibility into trends in student attendance 

# What does this PR do?
Implements a bare bones router so a user can navigate between the two dashboards
Creates a dashboard similar to the absence dashboard which shows the tardies for the past three months.

# Screenshot (if adding a client-side feature)
tardies_dashboard_screenshot

# Checklists

## Javascript QA

*If the PR touches Javascript, follow this checklist. If the PR doesn't touch Javascript, you can delete this section.*

*Which features or pages does this PR touch? Make a list here so that you can double-check each affected page in a Internet Explorer before merging.*

School Absence Dashboard
School Tardies Dashboard

+ [x] Author checked latest in IE - School Absence Dashboard
+ [x] Author checked latest in IE - School Tardies Dashboard
+ [x] Reviewer checked latest in IE - School Absence Dashboard
+ [x] Reviewer checked latest in IE - School Tardies Dashboard

*Does the PR introduce a new feature or new section of code?*

+ [x] Author included specs for new code

### CODE REVIEW

**Comment by alexsoble** [comment]
Hey @edavidsonsawyer! Looks like we still have two sub-issues open in #1264. We should fix those first before moving on to the next dashboard. It looks like both of them will apply to the tardies dashboard as well, so would be easier to implement once an re-use rather than implement twice. Let me know what you think!

**Comment by edavidsonsawyer** [comment]
Hey @alexsoble, I agree about the merge order. This is here because I wanted to work through some of these changes before adding the school year restriction in #1264. It seemed possible that would need to change again once these components were modified to accommodate more dashboards. It turns out that won't be a problem, so I should be able to merge this after finishing #1264. The tardy dashboard doesn't deal with date ranges, so the two open issues in #1264 wouldn't affect this feature.

File: app/assets/javascripts/school_administrator_dashboard/dashboard_components/school_administrator_dashboards.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import SchoolwideTardies from './tardies_dashboard/schoolwide_tardies.jsx';
+
+export default React.createClass({
+  displayName: 'SchoolAdministratorDashboards',
+
+  propTypes: {
+    serializedData: React.PropTypes.object.isRequired
+  },
+
+  render: function() {
```
Comment (kevinrobinson) [inline app/assets/javascripts/school_administrator_dashboard/dashboard_components/school_administrator_dashboards.jsx]: I'd suggest finishing off this PR and not solving this right now in this PR, but over time let's converge towards having routing all in one place.  This adds a second "layer" of routing beyond the main one in https://github.com/studentinsights/studentinsights/blob/master/ui/index.js#L27 and https://github.com/studentinsights/studentinsights/blob/master/ui/route.js.

Maybe let's keep moving and come back to "Convert JS routing to use React Router" in a separate issue later?

Comment (edavidsonsawyer) [reply app/assets/javascripts/school_administrator_dashboard/dashboard_components/school_administrator_dashboards.jsx]: @kevinrobinson I'm also open to using the existing router, but I got a little stuck. The problem I'm trying to solve is that I don't want to reload the data for the dashboards every time a user switches the view. If there's a straightforward way to do that without something like react-router, I'm happy to go that route. If you have some bright ideas and a few minutes for a call, I'd be glad to talk it through!

Comment (kevinrobinson) [reply app/assets/javascripts/school_administrator_dashboard/dashboard_components/school_administrator_dashboards.jsx]: @edavidsonsawyer Yeah I get you, and that seems great!  I think doing what you're doing here is a fine immediate-term solution, and no need to block and solve that differently now.

Another immediate-term solution could be having the component here hold a bit of state for which page should be shown, and then pass down a callback to the different dashboard components that let them change to another dashboard.  But I think what you're doing here is a good solution for now - my vote would be to just keep moving and ship it while calling out it's a thing to refactor later.

**Comment by edavidsonsawyer** [comment]
Hey, @alexsoble, @kevinrobinson or whoever else wants to review. This should be ready for a look!

**Comment by alexsoble** [comment]
Reviewing this now with @edavidsonsawyer! 

Eric noticed some funkiness in the absences categories on the x-axis: 

screen shot 2017-12-15 at 12 27 10 pm

We also decided to merge in datepicker branch first and then rebase this on top of it.

**Comment by edavidsonsawyer** [comment]
Alright, @alexsoble, this takes care of the absence dashboard bug and should be ready for review.

**Comment by alexsoble** [comment]
@edavidsonsawyer 👍 👍  — reviewing now

**Comment by alexsoble** [comment]
Hey there @edavidsonsawyer! I gave this a review in Internet Explorer, looks good! Let's merge, keeping in mind @kevinrobinson's longer-term notes about centralizing on one routing solution.

For future refactors/PRs in this area, I think it would be good to start moving these dashboard components over to the same React/ES6 style we're using for other components: using `import React from 'react'` and `class AwesomeClass extends React.Component`. When we add new components like SchoolTardiesDashboard, let's add them this way so that we don't have to refactor later down the line. 

For examples of the React ES6 module style I'm talking about, check out these PRs:

— https://github.com/studentinsights/studentinsights/pull/1346
— https://github.com/studentinsights/studentinsights/pull/1345
— https://github.com/studentinsights/studentinsights/pull/1344

Cheers and congrats on shipping this next dashboard @edavidsonsawyer! 🎉

**Comment by alexsoble** [comment]
@edavidsonsawyer Looks good in prod! 🎉

**Pull request merged**